### PR TITLE
Create/edit video integration

### LINKED ISSUE #369: Integrate video create/edit with blockchain
Once the basic functionality for video create/edit is done, we should allow doing actual blockchain transaction, display the dialog, etc. This will also require adding video-related logic to `joystream-js`. All of this should be very similar to what's done for the channel edit

### PR DESCRIPTION
**Pull request by drillprop**
Fix #369

### CODE REVIEW

File: src/joystream-lib/api.ts [hunk 1]
```diff
@@ -212,6 +222,123 @@ export class JoystreamJs {
...
+    if (inputMetadata.mediaPixelWidth != null) {
+      protoMeta.setMediaPixelWidth(inputMetadata.mediaPixelWidth)
+    }
+    if (inputMetadata.hasMarketing != null) {
+      protoMeta.setHasMarketing(inputMetadata.hasMarketing)
+    }
+
+    // todo figure out how to do it cleaner
+    if (inputMetadata.mediaType != null) {
+      const protoMediaType = protoMeta.getMediaType()
```
Comment (kdembler) [inline src/joystream-lib/api.ts]:
```ts
const protoMediaType = new MediaType()
....
protoMeta.setMediaType(protoMediaType)
```
?

File: src/joystream-lib/api.ts [hunk 2]
```diff
@@ -212,6 +228,122 @@ export class JoystreamJs {
...
+    }
+    if (inputMetadata.hasMarketing != null) {
+      protoMeta.setHasMarketing(inputMetadata.hasMarketing)
+    }
+
+    if (inputMetadata.mediaType != null) {
+      const protoMediaType = new MediaType()
+      protoMediaType.setCodecName(inputMetadata.mediaType.codecName || '')
+      protoMediaType.setContainer(inputMetadata.mediaType.container || '')
+      protoMediaType.setMimeMediaType(inputMetadata.mediaType.mimeMediaType || '')
```
Comment (kdembler) [inline src/joystream-lib/api.ts:285]: Your create `protoMediaType` instance but you don't use it at all? Unless you do `protoMeta.setMediaType(protoMediaType)` this won't have any effect

Comment (kdembler) [reply src/joystream-lib/api.ts:285]: Same for license and publishedBeforeJoystream

File: src/views/studio/EditVideoSheet/EditVideoSheet.tsx [hunk 3]
```diff
@@ -41,8 +69,37 @@ export const EditVideoSheet: React.FC = () => {
...
+  })
+  const [thumbnail, setThumbnail] = useState<Asset>({
+    url: null,
+    blob: null,
+  })
+  const {
+    activeUser: { channelId, memberId },
+  } = useActiveUser()
+
+  const { refetch: refetchMember } = useMembership(
```
Comment (kdembler) [inline src/views/studio/EditVideoSheet/EditVideoSheet.tsx]: Why do we care about refetching the membership here?

File: src/views/studio/EditVideoSheet/EditVideoSheet.tsx [hunk 4]
```diff
@@ -60,36 +117,195 @@ export const EditVideoSheet: React.FC = () => {
...
+      const { data: newVideoId, block } = await joystream.createVideo(
+        memberId,
+        channelId,
+        metadata,
+        assets,
+        (status) => {
+          setTransactionStatus(status)
+        }
+      )
+      assetsOwner = newVideoId
```
Comment (kdembler) [inline src/views/studio/EditVideoSheet/EditVideoSheet.tsx]: Channel is the owner of the assets, this is used to filter only assets for a given channel in the uploads view

File: src/views/studio/EditVideoSheet/EditVideoSheet.tsx [hunk 5]
```diff
@@ -60,36 +117,195 @@ export const EditVideoSheet: React.FC = () => {
...
+
+      setTransactionStatus(ExtrinsicStatus.Syncing)
+      setTransactionBlock(block)
+      setTransactionCallback(() => async () => {
+        await refetchMember()
```
Comment (kdembler) [inline src/views/studio/EditVideoSheet/EditVideoSheet.tsx]: Like mentioned, above, I don't think there's any reason to do that

**Review by kdembler — CHANGES_REQUESTED** [review]

File: src/joystream-lib/api.ts (hunk 2 above)
Comment (drillprop) [reply src/joystream-lib/api.ts:285]: 🤦🏻  right! Now everything works as it should 😅

File: src/views/studio/EditVideoSheet/EditVideoSheet.tsx [hunk 6]
```diff
@@ -60,36 +110,186 @@ export const EditVideoSheet: React.FC = () => {
...
+      const [asset, contentId] = joystream.createFileAsset({
+        size: thumbnail.blob.size,
+        ipfsContentId: await thumbnailHashPromise,
+      })
+      assets.thumbnail = asset
+      thumbnailContentId = contentId
+    } else {
+      console.warn('Missing thumbnail data')
+    }
+    let assetsOwner: VideoId = ''
```
Comment (kdembler) [inline src/views/studio/EditVideoSheet/EditVideoSheet.tsx]: We don't really need this anymore, right?

File: src/views/studio/EditVideoSheet/EditVideoSheet.tsx [hunk 7]
```diff
@@ -60,36 +110,186 @@ export const EditVideoSheet: React.FC = () => {
...
+          type: 'video',
+        })
+      }
+      if (thumbnail.blob && thumbnailContentId) {
+        startFileUpload(thumbnail.blob, {
+          contentId: thumbnailContentId,
+          owner: assetsOwner,
+          parentObject: {
+            type: 'video',
+            id: assetsOwner,
```
Comment (kdembler) [inline src/views/studio/EditVideoSheet/EditVideoSheet.tsx]: This needs the video ID :P

File: src/views/studio/EditVideoSheet/EditVideoSheet.tsx [hunk 8]
```diff
@@ -214,8 +437,44 @@ export const EditVideoSheet: React.FC = () => {
     })
     .filter((tab) => tab.title !== undefined)
 
+  const handleTransactionClose = async () => {
+    if (transactionStatus === ExtrinsicStatus.Completed) {
+      if (selectedVideoTab?.id) {
+        removeDraft(selectedVideoTab?.id)
+      }
+      setTransactionStatus(null)
+      closeSheet()
```
Comment (kdembler) [inline src/views/studio/EditVideoSheet/EditVideoSheet.tsx]: Let's minimize it here

**Review by kdembler — APPROVED** [review]

**Pull request merged**